### PR TITLE
fix(docs): fix spec.postgresql.ldap.tls field description

### LINKED ISSUE #587: Incorrect Docs\LDAPConfig tls description
Currenty documentation says:
<html>
<body>

tls | Set to 1 to enable LDAP over TLS | bool
-- | -- | --

</body>
</html>
which is incorrect, since tls option accepts only true value (false is the default).

### PR DESCRIPTION
**Pull request by MichaluxPL**
Closes #587

### CODE REVIEW

File: docs/src/api_reference.md [hunk 1]
```diff
@@ -568,7 +568,7 @@ Name           | Description
 `server        ` | LDAP hostname or IP address                                     | string                                    
 `port          ` | LDAP server port                                                | int                                       
 `scheme        ` | LDAP schema to be used, possible options are `ldap` and `ldaps` | LDAPScheme                                
-`tls           ` | Set to 1 to enable LDAP over TLS                                | bool                                      
+`tls           ` | Set to true to enable LDAP over TLS. false is default           | bool
```
Comment (gbartolini) [inline docs/src/api_reference.md:571]:
```suggestion
`tls           ` | Set to `true` to enable LDAP over TLS. `false` is default       | bool                                      
```

**Review by gbartolini — APPROVED** [review]

**Comment by phisco** [comment]
Hi @MichaluxPL, thanks for your contribution! Unfortunately, that part of the doc is autogenerated, so could you also update the string in the code [here](https://github.com/cloudnative-pg/cloudnative-pg/blob/main/api/v1/cluster_types.go#L657)?

**Review by jsilvela — APPROVED** [review]

**Comment by gbartolini** [comment]
@leonardoce rightly hinted that we need to change also `api/v1/cluster_types.go` and then run `make checks` to regenerate the .md file. @MichaluxPL can you please retry?

**Review by gbartolini — CHANGES_REQUESTED** [review]
Please change api/v1/cluster_types.go file and run make checks to regenerate this file.

**Comment by MichaluxPL** [comment]
> Please change api/v1/cluster_types.go file and run make checks to regenerate this file.

I've tried to run make checks, but it fails with error:
pkg/configparser/configparser.go:21: File is not `gofmt`-ed with `-s` (gofmt)
1. a struct containing the default values
2. a map from string to strings (e.g. the one that can be extracted for a Kubernetes ConfigMap)
3. an environment source like the system environment variables or a "fake"
   one suitable for testing.
pkg/utils/imagename.go:83: File is not `gofmt`-ed with `-s` (gofmt)
//     GetImageTag("postgres") == "latest"
//     GetImageTag("ghcr.io/cloudnative-pg/postgresql:12.3") == "12.3"
//
pkg/utils/pod_conditions.go:58: File is not `gofmt`-ed with `-s` (gofmt)
// IsPodActive checks if a pod is active, copied from:
controllers/pooler_update.go:57: File is not `gofumpt`-ed (gofumpt)

controllers/pooler_update.go:118: File is not `gofumpt`-ed (gofumpt)

make: *** [Makefile:154: lint] Error 1

**Comment by jsilvela** [comment]
@MichaluxPL if you do `make apidoc` that should do the changes.

**Comment by MichaluxPL** [comment]
> @MichaluxPL if you do `make apidoc` that should do the changes.

OK, I've pushed additional changes, You've mentioned.

File: Makefile [hunk 2]
```diff
@@ -151,7 +151,7 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 lint: ## Run the linter.
-	golangci-lint run
+	golangci-lint run --timeout 10m0s
```
Comment (phisco) [inline Makefile:154]: please remove this

Comment (MichaluxPL) [reply Makefile]: > please remove this

Removed - sorry about that ;)

**Review by phisco — APPROVED** [review]

**Review by sxd — APPROVED** [review]

**Review by phisco — APPROVED** [review]

**Review by jsilvela — APPROVED** [review]

**Pull request merged**